### DISCUSSION TITLE
How to login to sharepoint with command on linux machine

### DISCUSSION THREAD

**Discussion by maiductuan** [abraunegg/onedrive#1890] [Q&A] [answered]
How to login to sharepoint with command on linux machine, I can't remote into virtual machine, can only ssh
-----------------------------------------log-------------------------
```
Loading config ...
No config file found, using defaults
Initializing the OneDrive API ...
Authorize this app visiting:

Invalid uri
Could not initialize the OneDrive API
[https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=22c49a0d-d21c-4792-aed1-8f163c982546&scope=files.readwrite%20files.readwrite.all%20offline_access&response_type=code&redirect_uri=https://login.microsoftonline.com/common/oauth2/nativeclient](https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=22c49a0d-d21c-4792-aed1-8f163c982546&scope=files.readwrite%20files.readwrite.all%20offline_access&response_type=code&redirect_uri=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fnativeclient&fbclid=IwAR3PwBK7xRF7y4_V0DVG47wT8jPUAqXAnJEA5YBOmzE69_AbjTmxcqVi4gQ)

Enter the response uri:
ERROR: script returned exit code 1
Finished: FAILURE
```

**Comment by abraunegg** [comment, marked as answer]
@maiductuan 
The first issue is what 'onedrive' client version are you running - please provide the output of `onedrive --version`

> How to login to sharepoint with command on linux machine. I can't remote into virtual machine, can only ssh

You do not need remote graphical display. The authentication URL you are given, you can use on any other system | browser to obtain the response URI - which you can then paste back into your SSH connection where the application is waiting for your response.

You can also present the authentication response as a key value pair, where the application waits for the file containing your response, then reads that authentication response from that file:
```
--auth-files ARG
      Perform authorization via two files passed in as ARG in the format `authUrl:responseUrl`
      The authorization URL is written to the `authUrl`, then onedrive waits for the file `responseUrl`
      to be present, and reads the response from that file.
```

**Reply by maiductuan** [reply]
I am using version v1.0.1
Is the URL I need to enter my sharepoint address?

**Reply by abraunegg** [reply]
@maiductuan 
> I am using version v1.0.1

This is a **MAJOR** issue ... follow the following steps:

1. Uninstall this version following whatever process is necessary for you to do so. This may be to uninstall a package, perform a system clean up, remove systemd files - or a whole lot more.
2. Install the correct client for your system. You must be running version v2.4.16 or greater - any other version less than 2.4.16 is obsolete and not supported.

**Reply by maiductuan** [reply]
What is the command to install linux, please tell me

**Reply by abraunegg** [reply]
Please read https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md *very* carefully

**Reply by maiductuan** [reply]
Is there a simpler way to push files to Sharepoint?
eg: `curl --ntlm -u email@domain.com:pass --upload-file hi.txt http://company.sharepoint.com/sites/Web Projects/hi.txt`

**Reply by abraunegg** [reply]
@maiductuan 


> Is there a simpler way to push files to Sharepoint? eg: `curl --ntlm -u email@domain.com:pass --upload-file hi.txt http://company.sharepoint.com/sites/Web Projects/hi.txt`

No this is not possible to do this.
